docs/source/evaluation_scripts.md

```diff
@@ -129,3 +129,205 @@ Here are the steps to configure remote evaluation:
+
+In code-upload challenges, the evaluation is tighly-coupled with the agent and environment containers: The agent container continuously interacts with the environment container, and the environment container provides feedback to the agent until the environment provides a `stop` signal when the goal is reached or when the agent cannot continue further. Therefore, it is important to describe the interaction between the containers, how once can set up the individual containers, and how can the evaluation be performed.
+
+The starter templates for code-upload challenge evaluation can be found [here](https://github.com/Cloud-CV/EvalAI-Starters/tree/master/code_upload_challenge_evaluation). Here, we have a code organization to facilitate communication between the agent and environment.
+
+Here are the steps to configure evaluation for code upload challenges:
+
+1. **Create an environment**:
+    There are few steps involved in creating an environment:
+    1. *Edit the evaluator_environment*: This class defines the environment which can be like a [gym environment](https://www.gymlibrary.dev/content/environment_creation/) or a [habitat environment](https://github.com/facebookresearch/habitat-lab/blob/b1f2d4791a0065d0791001b72a6c96748a5f9ae0/habitat-lab/habitat/core/env.py) and other related attributes and method. Modify the `evaluator_environment` containing a gym environment shown [here](https://github.com/Cloud-CV/EvalAI-Starters/blob/8338085c6335487332f5b57cf7182201b8499aad/code_upload_challenge_evaluation/environment/environment.py#L21-L32):
+
+        ```python
+        class evaluator_environment:
+            def __init__(self, environment="CartPole-v0"):
+                self.score = 0
+                self.feedback = None
+                self.env = gym.make(environment)
+                self.env.reset()
+
+            def get_action_space(self):
+                return list(range(self.env.action_space.n))
+
+            def next_score(self):
+                self.score += 1
+        ```
+
+        The attribute `self.env` here is the environment, `self.feedback` is the observations from the sensors of the agent after performing an action and `self.score` stores the score at each step.
+
+        The `next_score` method shows how the score can be updated and shown towards the end of the evaluation and the `get_action_space` gets the list of possible actions. You can add custom methods and attributes which help in interaction with the environment.
+
+    2. _Edit the Environment service_: This service is hosted on the GRPC server in order to get actions in form of messages from the agent container. Modify the lines shown [here](https://github.com/Cloud-CV/EvalAI-Starters/blob/8338085c6335487332f5b57cf7182201b8499aad/code_upload_challenge_evaluation/environment/environment.py#L35-L65):
```

Comment:
GRPC is just the communication protocal right? We don't run environment on the "GRPC server" if I understand correctly

Reply:
I was trying to explain [this line](https://github.com/Cloud-CV/EvalAI-Starters/blob/8338085c6335487332f5b57cf7182201b8499aad/code_upload_challenge_evaluation/environment/environment.py#L129) in the code:
```python
server = grpc.server(futures.ThreadPoolExecutor(max_workers=1))
```
I don't think it is a protocol, but rather a framework used for communication. I might have mentioned it as a protocol somewhere, but I will update that and provide a link to [GRPC](https://grpc.io/) the first time I mention it.